### PR TITLE
Commit with changes (fixes #9)

### LINKED ISSUE #9: Add to codebase reflection
- Add reflection to some methods;
- Refactor entities with new annotations;
- Refactor method for reading;

### PR DESCRIPTION
**Pull request by mkgerasimenko**
- refactored readFrom method;
 - added reflection to XmlReader class;
 - common refactoring.

### CODE REVIEW

File: src/main/java/com/waverleysoftware/data/impl/XmlReader.java [hunk 1]
```diff
@@ -22,10 +24,18 @@ public String getEntityType() {
...
-    public <T> T readFrom(final String dataSource, final Class<T> entityClass) {
-        return Try.of(() -> newInstance(entityClass))
-                .mapTry(JAXBContext::createUnmarshaller)
-                .mapTry(unmarshaller -> (T) unmarshaller.unmarshal(new File(getSystemResource(dataSource).getPath())))
+    public <T> T[] readFrom(final String dataSource, final Class<T> entityClass) {
+
+        return (T[]) Try.of(() -> new DOMBuilder().build(newInstance().newDocumentBuilder()
+                .parse(new File(getSystemResource(dataSource).getPath())))
+                .getRootElement()
+                .getChildren("Client"))
```
Comment (sskorol) [inline src/main/java/com/waverleysoftware/data/impl/XmlReader.java]: it's not a generic method anymore, as you've hardcoded exact entity nodes...

File: src/test/java/com/waverleysoftware/DataFromFileTest.java [hunk 2]
```diff
@@ -8,34 +8,28 @@
...
-import static org.joor.Reflect.on;
 
 /**
  * A simple class for some data testing.
  */
 @Slf4j
 public class DataFromFileTest extends BaseTest {
 
-    @Data(source = "panel.json", dataClass = Panel.class)
+    @Data(source = "panel.json", entity = Panel[].class)
```
Comment (sskorol) [inline src/test/java/com/waverleysoftware/DataFromFileTest.java]: would it work with a single entity class?

File: src/test/java/com/waverleysoftware/DataFromFileTest.java [hunk 3]
```diff
@@ -8,34 +8,28 @@
...
 
-    @Data(source = "client.xml", dataClass = Client.class)
+    @Data(source = "clientBase.xml", entity = Client.class)
     @Test(dataProvider = "getData", dataProviderClass = DataSuppliers.class)
     public void clientShouldBeModified(final Client client) {
         assertThat(client).hasJob("Child");
     }
 
-    @Data(source = "client.xml", dataClass = Client.class)
+    @Data(source = "clientBase.xml", entity = Client.class)
```
Comment (sskorol) [inline src/test/java/com/waverleysoftware/DataFromFileTest.java]: add more tests, which cover both single and multiple entities for json and xml

**Review by sskorol — CHANGES_REQUESTED** [review]
@mkgerasimenko check comments please

**Comment by mkgerasimenko** [comment]
@sskorol, please, look at the changes. Thank you.

**Comment by mkgerasimenko** [comment]
@sskorol, please, look at the changes. Thanks.

**Review by sskorol — APPROVED** [review]

**Pull request merged**